Tests/Builder/AbstractFormContractorTest.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Tests\Builder;
+
+/**
+ * @author ju1ius <ju1ius@laposte.net>
+ */
+class AbstractFormContractorTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @param string $formType
+     * @param string $targetEntity
+     * @param string $associationType
+     * @param string $associationAdmin
+     * @param string $expectedException
+     *
+     * @dataProvider testFieldDescriptionValidationProvider
+     */
+    public function testFieldDescriptionValidation(
+        $formType,
+        $targetEntity,
+        $associationType,
+        $associationAdmin,
+        $expectedException
+    ) {
```

Comment:
I wish I could avoid that "absolutely all text, be it in js, php, css, whatever, must be under 120 chars, well except class and method signatures that are sort of special because <insert weak reason here>" exception to an otherwise beautiful rule, trust me. Please don't get me started on this, I can rant on that all afternoon.